### PR TITLE
feat(input group):adds input group

### PR DESCRIPTION
**Pull request by matthewcarleton**
This creates the input group. 

A few things to note: 

I've also updated the the dropdown with the new spacers provided by design.

@mceledonia I've updated the input to be 36px. I don't know if we discussed if it was 36px with or without the 1px borders.

Please have a look and let me know what this is missing. I think I've taken the right approach with A11y but would love some feedback on that @jgiardino.

Thanks!

### CODE REVIEW

File: src/patternfly/components/InputGroup/examples/input-group-example.hbs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  {{#> form-controls controlType="input" input="true" type="text" id="textInput1" placeholder="Standard input example" aria-label="standard input example" }}
+  {{/form-controls}}
+  {{#> button btnClass="pf-m-secondary"}} Button {{/button}}
+  {{#> dropdown id="dropdown-example-collapsed"}}Dropdown{{/dropdown}}
+</div>
+<br>
+<br>
+<div class="pf-c-input-group">
+  <span class="pf-c-input-group__text">
+    <i class="fas fa-dollar-sign"></i>
```
Comment (mattnolting) [inline src/patternfly/components/InputGroup/examples/input-group-example.hbs]: I think `aria-label` is needed here.

File: src/patternfly/components/InputGroup/examples/input-group-example.hbs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  </span>
+</div>
+<br>
+<br>
+<div class="pf-c-input-group">
+  <span class="pf-c-input-group__text" id="inputGroupText1">
+    <i class="fas fa-at"></i>
+  </span>
+  {{#> form-controls controlType="input" input="true" type="email" id="textInput1" placeholder="Standard input example" aria-label="standard input example" aria-describedby="inputGroupText1" }}
+{{/form-controls}}
```
Comment (mattnolting) [inline src/patternfly/components/InputGroup/examples/input-group-example.hbs]: Not sure how browsers handle inputs describedby icons, even when/if they have a label. Probably best to drop `aria-describedby` in this case.

File: src/patternfly/components/InputGroup/examples/input-group-example.hbs [hunk 3]
```diff
@@ -0,0 +1,59 @@
+<div class="pf-c-input-group">
+  {{#> button btnClass="pf-m-secondary"}} Button {{/button}}
+  {{#> form-controls controlType="textarea" name="textarea" id="textarea1" aria-label="textarea example"}}
+{{/form-controls}}
+    {{#> button btnClass="pf-m-tertiary"}} Button {{/button}}
+</div> 
+<br>
+<br>
```
Comment (mattnolting) [inline src/patternfly/components/InputGroup/examples/input-group-example.hbs]: I imagine @andresgalante would say, scope workspace styling for spacing instead of adding `<br>`s. Not a blocker.

File: src/patternfly/components/Form/styles.scss [hunk 4]
```diff
@@ -227,7 +227,7 @@ $pf-c-form__select--m-error-svg--Viewbox: "0 0 26 15.8" !default;
     color: var(--pf-c-form__input--Color);
     background-color: var(--pf-c-form__input--BackgroundColor);
     border: var(--pf-c-form__input--BorderSize) solid var(--pf-c-form__input--BorderColor);
-    border-radius: 0;
+    border-radius: 3px;
```
[the remark this answers is not in the document]
Comment (matthewcarleton) [reply src/patternfly/components/Form/styles.scss]: ya, that shouldn't even be there. It will all get removed when I put up the form PR

File: src/patternfly/components/InputGroup/examples/input-group-example.hbs (hunk 3 above)
Comment (matthewcarleton) [reply src/patternfly/components/InputGroup/examples/input-group-example.hbs]: ya I've used this approach in alerts as well. Do you think we should break it out into separate examples?

File: src/patternfly/components/InputGroup/examples/input-group-example.hbs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  {{#> form-controls controlType="input" input="true" type="text" id="textInput4" name="textInput4" placeholder="Standard input example" aria-label="standard input example" }}
+  {{/form-controls}}
+  {{#> button btnClass="pf-m-secondary"}} Button {{/button}}
+  {{#> dropdown id="dropdown-example-collapsed"}}Dropdown{{/dropdown}}
+</div>
+<br>
+<br>
+<div class="pf-c-input-group">
+  <span class="pf-c-input-group__text">
+    <i class="fas fa-dollar-sign" aria-label="text input label"></i>
```
Comment (mattnolting) [inline src/patternfly/components/InputGroup/examples/input-group-example.hbs]: `aria-label` needs to be on span, as the icon is rendered with aria-hidden="true" same as button example

File: src/patternfly/components/InputGroup/examples/input-group-example.hbs [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+{{/form-controls}}
+  <span class="pf-c-input-group__text">
+    @example.com
+  </span>
+</div>
+<br>
+<br>
+<div class="pf-c-input-group">
+  <span class="pf-c-input-group__text">
+    <i class="fas fa-at" aria-label="text input label"></i>
```
Comment (mattnolting) [inline src/patternfly/components/InputGroup/examples/input-group-example.hbs]: `aria-label` needs to be on span, as the icon is rendered with aria-hidden="true" same as button example

File: src/patternfly/components/InputGroup/examples/input-group-example.hbs (hunk 3 above)
Comment (mattnolting) [reply src/patternfly/components/InputGroup/examples/input-group-example.hbs]: No, I think it's a good approach

File: src/patternfly/components/InputGroup/examples/input-group-example.hbs (hunk 5 above)
Comment (matthewcarleton) [reply src/patternfly/components/InputGroup/examples/input-group-example.hbs]: ha, that's what I had and I went with this method after I doubted myself :(

**Comment by mattnolting** [comment]
Looks good to me!

File: src/patternfly/components/Dropdown/styles.scss [hunk 7]
```diff
@@ -3,7 +3,7 @@
 .pf-c-dropdown {
   // Toggle
   --pf-c-dropdown__toggle--PaddingTop: var(--pf-global--spacer--xs);
-  --pf-c-dropdown__toggle--PaddingRight: var(--pf-global--spacer--md);
+  --pf-c-dropdown__toggle--PaddingRight: var(--pf-global--spacer--sm);
```
Comment (jgiardino) [inline src/patternfly/components/Dropdown/styles.scss]: Is this change intentional? Did the visual design for dropdowns change?

This is an example of the Dropdown before this change:
image
And after:
image

File: src/patternfly/components/InputGroup/examples/input-group-example.hbs [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+<div class="pf-c-input-group">
+  {{#> form-controls controlType="textarea" name="textarea2" id="textarea2" aria-label="textarea example"}}
+{{/form-controls}}
+    {{#> button btnClass="pf-m-tertiary"}} Button {{/button}}
+</div> 
+<br>
+<br>
+<div class="pf-c-input-group">
+{{#> dropdown id="dropdown-example-collapsed"}}Dropdown{{/dropdown}}
+  {{#> form-controls controlType="input" input="true" type="text" id="textInput3" name="textInput3" placeholder="Standard input example" aria-label="standard input example" }}
```
Comment (jgiardino) [inline src/patternfly/components/InputGroup/examples/input-group-example.hbs]: Since `placeholder` text is intended to provide hints about the expected value, then we should avoid using it in our examples for anything but it's intended purpose. One of the biggest usability/accessibility issues related to `placeholder` is that it's misused as a label for the input field, and using it here in this way might help to perpetuate this mis-usage.

File: src/patternfly/components/Dropdown/styles.scss (hunk 7 above)
Comment (matthewcarleton) [reply src/patternfly/components/Dropdown/styles.scss]: yes, it changed when @mceledonia updated the forms.

File: src/patternfly/components/InputGroup/examples/input-group-example.hbs (hunk 8 above)
Comment (matthewcarleton) [reply src/patternfly/components/InputGroup/examples/input-group-example.hbs]: that makes a lot of sense.

File: src/patternfly/components/InputGroup/docs/code.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
+## Overview
+Use the input group to extend form controls by adding text, buttons, dropdowns, etc.
+
+## Accessibility
+When using the `pf-c-input-group` always ensure labels are used outside the input group with the `.sr-only` class applied. You can also make use of the `aria-describedby`, `aria-label`, or `aria-labelledby` attributues. For more information on accessibility and forms see the <a href="/components/form/examples/">form component</a>.
```
Comment (andresgalante) [inline src/patternfly/components/InputGroup/docs/code.md]: there is no `.sr-only` class on patternfly

**Comment by andresgalante** [comment]
@kybaker @mceledonia are you guys ok with how this looks? https://deploy-preview-444--pf-next.netlify.com/components/InputGroup/examples/
